astropy/time/core.py

```diff
@@ -1468,12 +1469,21 @@ def _apply(self, method, *args, format=None, **kwargs):
                              .format(list(tm.FORMATS)))
 
         NewFormat = tm.FORMATS[new_format]
+
         tm._time = NewFormat(tm._time.jd1, tm._time.jd2,
                              tm._time._scale, tm.precision,
                              tm.in_subfmt, tm.out_subfmt,
                              from_jd=True)
         tm._format = new_format
         tm.SCALES = self.SCALES
+
+        # If current output subformat does not match any subfmts in new format
```

Comment:
Can we put this in an `if new_format is not self.format:` clause to avoid slowing down regular operations like indexing and reshaping?

Reply:
Done.  I used `!=` not `is not` since the user might supply a `format` string which is the same value as the current format.